pkg/cfn/builder/api_test.go

```diff
@@ -0,0 +1,229 @@
+			"VPC":                      "vpc-0e265ad953062b94b",
+			"Endpoint":                 "https://DE37D8AFB23F7275D2361AD6B2599143.yl4.us-west-2.eks.amazonaws.com",
+			"CertificateAuthorityData": caCert,
+			"ARN":              "arn:aws:eks:us-west-2:376248598259:cluster/ferocious-mushroom-1532594698",
+			"ClusterStackName": "",
+		})
+
+		It("should not error", func() {
+			err := rs.GetAllOutputs(sampleStack)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should be equal", func() {
+			Expect(initial).To(Equal(expected))
+		})
+	})
+
+	Describe("AutoNameTag", func() {
+		rs := NewNodeGroupResourceSet(&api.ClusterConfig{
+			AvailabilityZones: testAZs,
+		})
+		rs.AddAllResources()
+
+		template, err := rs.RenderJSON()
+		It("should serialise JSON without errors", func() {
+			Expect(err).NotTo(HaveOccurred())
+		})
+		obj := Template{}
+		It("should parse JSON withon errors", func() {
+			err := json.Unmarshal(template, &obj)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("SG should have correct tags", func() {
+			Expect(len(obj.Resources)).ToNot(Equal(0))
+			Expect(len(obj.Resources["SG"].Properties.Tags)).To(Equal(2))
+			Expect(obj.Resources["SG"].Properties.Tags[0].Key).To(Equal(map[string]interface{}{
+				"Fn::Sub": "kubernetes.io/cluster/${ClusterName}",
+			}))
+			Expect(obj.Resources["SG"].Properties.Tags[0].Value).To(Equal("owned"))
+			Expect(obj.Resources["SG"].Properties.Tags[1].Key).To(Equal("Name"))
+			Expect(obj.Resources["SG"].Properties.Tags[1].Value).To(Equal(map[string]interface{}{
+				"Fn::Sub": "${AWS::StackName}/SG",
+			}))
+		})
+	})
+
+	Describe("NodeGroupTags", func() {
+		rs := NewNodeGroupResourceSet(&api.ClusterConfig{
+			AvailabilityZones: testAZs,
+		})
+		rs.AddAllResources()
+
+		template, err := rs.RenderJSON()
+		It("should serialise JSON without errors", func() {
+			Expect(err).NotTo(HaveOccurred())
+		})
+		obj := Template{}
+		It("should parse JSON withon errors", func() {
+			err := json.Unmarshal(template, &obj)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("NodeGroup should have correct tags", func() {
+			Expect(len(obj.Resources)).ToNot(Equal(0))
+			Expect(len(obj.Resources["NodeGroup"].Properties.Tags)).To(Equal(2))
+			Expect(obj.Resources["NodeGroup"].Properties.Tags[0].Key).To(Equal("Name"))
+			Expect(obj.Resources["NodeGroup"].Properties.Tags[0].Value).To(Equal(map[string]interface{}{
+				"Fn::Sub": "${ClusterName}-${NodeGroupID}-Node",
+			}))
+			Expect(obj.Resources["NodeGroup"].Properties.Tags[0].PropagateAtLaunch).To(Equal("true"))
+			Expect(obj.Resources["NodeGroup"].Properties.Tags[1].Key).To(Equal(map[string]interface{}{
+				"Fn::Sub": "kubernetes.io/cluster/${ClusterName}",
+			}))
+			Expect(obj.Resources["NodeGroup"].Properties.Tags[1].Value).To(Equal("owned"))
+			Expect(obj.Resources["NodeGroup"].Properties.Tags[1].PropagateAtLaunch).To(Equal("true"))
+		})
+	})
+
+	Describe("UserData", func() {
```

Comment:
This mostly tests cloudconfig package, but in the context of how we actually use it.